core/arch/arm/kernel/generic_boot.c

```diff
@@ -1094,6 +1095,8 @@ static void discover_nsec_memory(void)
 
 	if (fdt) {
 		mem = get_memory(fdt, &nelems);
+		carve_out_phys_mem(&mem, &nelems, TZDRAM_BASE,
```

Comment:
How about doing this inside `check_phys_mem_is_outside()` in the same way as we're dealing with `MEM_AREA_NSEC_SHM`? We should probably have a list of memory types to carve out, `MEM_AREA_SDP_MEM` for instance.

Reply:
Hmm, I see that we're already handling SDP memory `check_phys_mem_is_outside()`.

Reply:
You probably mean `core_mmu_set_discovered_nsec_ddr()`? This is already done there. If we carve out all appropriate memory areas there, there is no real point in checking whether they are outside, since they are all carved out anyway. Than we might as well remove the check. The real problem here is that barebox always sets the complete memory in the memory dt node, while U-Boot will substract the OP-TEE memory apparently beforehand. Additionally I am running into issues with platforms which are not on the generic RAM layout yet, either because of redefinitions or because of undefined variables.

Reply:
Yes, I mean `core_mmu_set_discovered_nsec_ddr()`, sorry.

If there's no need to check, the better, then it's fail safe.

Reply:
I have pushed a new version which removes the device tree memory parsing, since OP-TEE essentially revalidates it's on configuration in the DT case.